sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/specialized/DataLakeLeaseClient.java

```diff
@@ -100,7 +105,12 @@ public String acquireLease(int duration) {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<String> acquireLeaseWithResponse(int duration, RequestConditions modifiedRequestConditions,
         Duration timeout, Context context) {
-        return blobLeaseClient.acquireLeaseWithResponse(duration, modifiedRequestConditions, timeout, context);
+        try {
```

Comment:
Can you use the returnOrMapException method in this file? Also, I think when you're constructing the DataLakeStorageException, it might be better to use one of the constructors that accepts a Throwable and pass the original exception in. I think this helps preserve the cause/stack trace

Reply:
Changed to use returnOrMapException

Reply:
For some reason that constructor doesn't work